### ISSUE TITLE
[no-let] Add option to ignore lines based on search string

### ISSUE THREAD

**Issue by realmikkifriend** [eslint-functional/eslint-plugin-functional#987]
Thanks very much to the devs for this library, which has helped me understand functional programming more than any tutorial.

## Background
Svelte passes props between components using a different method depending on which major version is used:
* `export let variable` ([Svelte 4, what I'm using](https://svelte.dev/docs/svelte/legacy-export-let))
* `let { variable } = $props();` ([Svelte 5, based on documentation](https://svelte.dev/docs/svelte/v5-migration-guide#Reactivity-syntax-changes-export-let-$props), see [discussion](https://github.com/sveltejs/svelte/issues/12121))

In addition to this plugin, I enabled ESLint to scan Svelte components in my codebase by installing `eslint-plugin-svelte` and enabling the recommended configs per documentation.

## Issue
Either valid usage of `let` in Svelte components triggers the `no-let` rule. I can't find a way to prevent this using this library's existing options, `eslint-plugin-svelte`, or ESLint configuration.

    export let variable; // valid Svelte 4 syntax, incorrectly shows no-let error

    let { variable } = $props(); // valid Svelte 5 syntax, incorrectly shows no-let error

    let variable; // correctly shows no-let error

For reference, while I don't want to lose the benefit of `no-let` linting, it's possible to disable this rule in Svelte files in `eslint.config.mjs`:

    {
        files: ["**/*.svelte"],
        rules: {
            "functional/no-let": "off",
        },
    },

## Suggestion
In addition to the `ignoreIdentifierPattern` option, it would be great if the `no-let` rule had an `ignoreLinePattern` (or `ignoreCodePattern` for consistency with rules like `no-classes`) option that skips matching lines. It would accept either search strings or arrays of search strings (e.g. `^export ` and `\$props\(\)`).

I barely understand ESLint but have searched and prompted for a way to configure it like this. As far as I can tell, ESLint configuration blocks (like above) can't disable linting by search string. I would love for this to be incorrect, because it would bypass the need to add new features.

**Comment by RebeccaStevens** [comment]
Svelte is one of those things I've been meaning to look into for ages but still haven't gotten around to it.

I assume there is a reason you can't simply use `const` instead of `let`, but I'm not aware of it. Could you enlighten me?

```ts
export const variable;
const { variable } = $props();
```

**Comment by realmikkifriend** [comment]
Great question...

## Svelte 4

Changing `export let variable` to `export const variable` breaks my app in ways that I couldn't immediately chase down/refactor. There also isn't documentation or much discussion of this use. Svelte 4 assumes you will use `export let variable` in a component receiving `variable` as a prop.

## Svelte 5

I can't test this myself in my current project/workspace, but [per this discussion](https://github.com/sveltejs/svelte/issues/12121), `const { value } = $props()` is also valid.

However, almost all Svelte 5 documentation uses `let` in this case, I think to acknowledge that the value is reactive and will therefore probably change. I personally think of reactivity as separate from functional patterns (which I'm still learning), so I would use `const` for it. But I wonder if other Svelte devs will be confused over this (if so, I hope they find this issue).
